### PR TITLE
fix(commandxlat): Consume the cheat and demo messages consistently when handled

### LINKED ISSUE #2769: GameMessage::MSG_CHEAT_ADD_CASH in CommandXlat does not consume message
```cpp
    case GameMessage::MSG_CHEAT_ADD_CASH:
		{
			if ( !TheGameLogic->isInMultiplayerGame() )
			{
				Player *localPlayer = ThePlayerList->getLocalPlayer();
				Money *money = localPlayer->getMoney();
				money->deposit( 10000 );
				TheInGameUI->messageNoFormat( TheGameText->FETCH_OR_SUBSTITUTE("GUI:DebugAddCash", L"Add Cash") );
			}
			break;
		}
```

`MSG_CHEAT_ADD_CASH` is the only cheat case in this block that does not set `disp = DESTROY_MESSAGE` after acting. Without it the message continues propagating through the pipeline when not in multiplayer; all neighboring cases (`MSG_CHEAT_INSTANT_BUILD`, `MSG_CHEAT_GIVE_ALL_SCIENCES`, etc.) consistently destroy the message.

### PR DESCRIPTION
**Pull request by bas-slats**
* Fixes #2769

`CommandTranslator::translateGameMessage` has several cheat and demo cases that perform their action without consuming the message, unlike their sibling cases, which set `disp = DESTROY_MESSAGE` when handled. This change consumes the message in all of them:

- `MSG_CHEAT_ADD_CASH` (the issue's subject)
- `MSG_CHEAT_SHOW_HEALTH`
- `MSG_META_DEMO_ADD_CASH`
- `MSG_META_DEMO_TOGGLE_RENDER`
- `MSG_META_DEMO_KILL_AREA_SELECTION`
- `MSG_META_DEMO_TOGGLE_METRICS` (`PERF_TIMERS` builds)
- `MSG_META_DEMO_PERFORM_STATISTICAL_DUMP` (`DUMP_PERF_STATS` builds)

Found by sweeping the whole switch for cheat/demo cases that lack a `disp` assignment; the stacked `RUNSCRIPT` and `OBJECTIVE_MOVIE` blocks and the remaining cheat/demo cases already consume. Code comments are removed per review.

## Testing

Compiles with the `win32` preset (VS2019 16.11, Ninja Multi-Config, Release); `CommandXlat.cpp` builds without warnings and `generalszh.exe` links. The two `#ifdef`-guarded sites are not compiled by this preset and are syntax-identical one-liners; the debug/profile CI presets exercise them.

An earlier build of this branch (with the two add cash changes) was smoke tested in game against retail Steam Zero Hour 1.04 data: boots, loads and plays a skirmish normally. The cheat messages are not bound in normal builds, so the disposition change is verified by inspection against the sibling cases.

---

This change was developed with AI assistance (Claude), human-directed: every case in the switch was checked by hand for the missing disposition, and the change was verified to compile as above. No gameplay simulation code is touched.

### CODE REVIEW

**Comment by Skyaero42** [comment]
There are more places where the message isn't consumed, like `MSG_META_DEMO_TOGGLE_RENDER`. Shouldn't it be consumed there as well?

File: Core/GameEngine/Source/GameClient/MessageStream/CommandXlat.cpp [hunk 1]
```diff
@@ -3677,6 +3677,9 @@ GameMessageDisposition CommandTranslator::translateGameMessage(const GameMessage
 				Money *money = localPlayer->getMoney();
 				money->deposit( 10000 );
 				TheInGameUI->messageNoFormat( TheGameText->FETCH_OR_SUBSTITUTE("GUI:DebugAddCash", L"Add Cash") );
+
+				// TheSuperHackers @fix Consume the message consistently with the other cheat messages.
```
Comment (xezon) [inline Core/GameEngine/Source/GameClient/MessageStream/CommandXlat.cpp]: Needs no comment. Debug functionality. No one will look back at it.

**Comment by xezon** [comment]
> There are more places where the message isn't consumed, like MSG_META_DEMO_TOGGLE_RENDER. Shouldn't it be consumed there as well?

Yes looks like there are a couple more.

**Comment by bas-slats** [comment]
Swept the whole `translateGameMessage` switch for cheat and demo cases that perform their action without setting a disposition. Five more cases consume the message now: `MSG_CHEAT_SHOW_HEALTH`, `MSG_META_DEMO_TOGGLE_RENDER`, `MSG_META_DEMO_KILL_AREA_SELECTION`, `MSG_META_DEMO_TOGGLE_METRICS` (`PERF_TIMERS` builds) and `MSG_META_DEMO_PERFORM_STATISTICAL_DUMP` (`DUMP_PERF_STATS` builds). The stacked `RUNSCRIPT` and `OBJECTIVE_MOVIE` blocks and the remaining cheat/demo cases already consume. The code comments are removed.

File: Core/GameEngine/Source/GameClient/MessageStream/CommandXlat.cpp [hunk 2]
```diff
@@ -4913,6 +4920,7 @@ GameMessageDisposition CommandTranslator::translateGameMessage(const GameMessage
 					obj->kill();
 				}
 			}
+			disp = DESTROY_MESSAGE;
```
Comment (Skyaero42) [inline Core/GameEngine/Source/GameClient/MessageStream/CommandXlat.cpp:4923]: nit: white line after closing bracket.
Two times in this file.

Comment (xezon) [inline Core/GameEngine/Source/GameClient/MessageStream/CommandXlat.cpp:4923]: Maybe only destroy if something was killed? Not sure if that makes sense.

Comment (Caball009) [reply Core/GameEngine/Source/GameClient/MessageStream/CommandXlat.cpp:4923]: Seems fine to me to destroy the message either way.

**Comment by Caball009** [comment]
@bas-slats Looks like the PR is almost ready, but there are still unresolved review points.

File: Core/GameEngine/Source/GameClient/MessageStream/CommandXlat.cpp (hunk 2 above)
Comment (xezon) [reply Core/GameEngine/Source/GameClient/MessageStream/CommandXlat.cpp:4923]: It means the key mapping will be consumed here and usable for nothing else afterwards, despite taking no action.
